src/platforms/common/jtagtap.c

```diff
@@ -51,8 +53,12 @@ void jtagtap_init(void)
 	jtag_proc.tap_idle_cycles = 1;
 
 	/* Ensure we're in JTAG mode */
+#if 1
+	jtagtap_cycle(true, false, 51U);
```

Comment:
Updated as such. Note that it would be `ADIV5_SWD_TO_JTAG_SELECT_SEQUENCE` if the unit included adiv5.h -- this magic number also occurs in hosted/ftdi_jtag. That one will use `_next()`.
I assume the `jtagtap_cycle()` method is a somewhat recent addition -- it's not used widely across the codebase, if only in BMPremote. I can even eliminate it as dead code for ADIv5 and RISC-V DTM targets, breaking the v0/v2 remote contract; unsure about AVR PDI. It's also too fast in comparison to other methods.

Reply:
As we've got a `#define` for it in that header, include the header - that's fine. We should look to reduce the number of places this constant is defined down to the one, documented, and global place ideally.

Correct, `jtagtap_cycle()` was introduced in v1.8 and the BMP remote protocol implementation may need tweaking to provide a jtagtap_next()-based fallback, but we're principally fine with that as a pivot as this is exactly the kind of thing it was introduced for from what we can tell from Uwe's PR and commit for it.

Reply:
We went checking and the remote protocol has this fallback - apparently we thought about this when rewriting BMDA's implementation. We should verify fallbacks for the other adaptors are present, but as most of them outright replace this init code we don't see that being a problem as such - though it is a target for potential future dedupe in the backends.